src/rpc/core_rpc_server.cpp

```diff
@@ -118,6 +169,42 @@ namespace cryptonote
     if (!rpc_config)
       return false;
 
+    std::string address = command_line::get_arg(vm, arg_rpc_payment_address);
+    if (!address.empty())
+    {
+      cryptonote::address_parse_info info;
+      if (!get_account_address_from_str(info, nettype(), address))
+      {
+        MERROR("Invalid payment address: " << address);
+        return false;
+      }
+      if (info.is_subaddress)
+      {
+        MERROR("Payment address may not be a subaddress: " << address);
+        return false;
+      }
+      uint64_t diff = command_line::get_arg(vm, arg_rpc_payment_difficulty);
+      uint64_t credits = command_line::get_arg(vm, arg_rpc_payment_credits);
+      if (diff == 0 || credits == 0)
+      {
+        MERROR("Payments difficulty and/or payments credits are 0, but a payment address was given");
+        return false;
+      }
+      m_rpc_payment.reset(new rpc_payment(info.address, diff, credits));
+      m_rpc_payment->load(command_line::get_arg(vm, cryptonote::arg_data_dir));
+      m_p2p.set_rpc_credits_per_hash(RPC_CREDITS_PER_HASH_SCALE * (credits / (float)diff));
+    }
+
+    if (!m_rpc_payment)
+    {
+      uint32_t bind_ip;
+      bool ok = epee::string_tools::get_ip_int32_from_string(bind_ip, rpc_config->bind_ip);
+      if (ok & !epee::net_utils::is_ip_loopback(bind_ip))
+        MWARNING("The RPC server is accessible from the outside, but no RPC payment was setup. RPC access will be free for all.");
+    }
+
+    m_local_client = rct::rct2sk(rct::skGen());
```

Comment:
I don't think so, not 100% sure though.
If the local daemon bootstraps on a remote one, the RPC are proxied, so the wallet will mine to the remote daemon. The wallet has a --rpc-client-secret-key argument which allows it to reuse the same key between runs (at the cost of privacy). Since the whole system is meant to mine a bit ahead and not a huge amount, it's not such a problem if balance is lost, it's really supposed to be lost and not care much.

Reply:
To expand on this, this local client is used when the deamon runs its own RPC, like "monerod status" on a detached daemon.